### PR TITLE
Add '.example' files.

### PR DESCRIPTION
**Pull request by davebuk**
Ref #2232 add, add platformio_override.ini.example and custom.h.example files.

### CODE REVIEW

File: code/espurna/config/custom.h.example [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+// See 'code/espurna/config/general.h' for default settings.
+//
+// See: https://github.com/xoseperez/espurna/wiki/Software-features#enabling-features
+// and 'code/platformio_override.ini.example' for more details.
+// ------------------------------------------------------------------------------
+
+
+// Example of a Sonoff Basic board with some options disabled to reduce firmware size.
+
+#if defined(MY_SONOFF_BUILD01)    // This matches the 'src_build_flags = -DMY_SONOFF_BUILD01'
```
Comment (mcspr) [inline code/espurna/config/custom.h.example]: *eaasl, maybe I misunderstand what 'matches' means here*

"This section will be used when `src_build_flags` contains -DMY_SONOFF_BUILD01"?

File: code/platformio_override.ini.example [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+# own file: 'platformio_override.ini'
+# 'platformio_override.ini' allows users to create their own environments to add, remove or change
+# firmware configurations. 
+#
+# See: https://github.com/xoseperez/espurna/wiki/PlatformIO#customize-build-settings
+# for more details.
+# -------------------------------------------------------------------------------------------------------------
+
+// Example for a standard Sonoff Basic R2 or R3 but enabling DHT support on GPIO 2.
+// It can be built from VSCode/PlatformIO by selecting the 'my-sonoff-build-dht'
```
Comment (mcspr) [inline code/platformio_override.ini.example]: "selecting ... build selection" -> "selecting the 'Build: my-sonoff-build-dht'"?

File: code/platformio_override.ini.example [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+# Either copy and paste this file, then rename removing the .example or create your
+# own file: 'platformio_override.ini'
+# 'platformio_override.ini' allows users to create their own environments to add, remove or change
+# firmware configurations. 
+#
+# See: https://github.com/xoseperez/espurna/wiki/PlatformIO#customize-build-settings
+# for more details.
+# -------------------------------------------------------------------------------------------------------------
+
+// Example for a standard Sonoff Basic R2 or R3 but enabling DHT support on GPIO 2.
```
Comment (mcspr) [inline code/platformio_override.ini.example]: afaik, there are only 2 allowed comment prefixes - `;` and `#`. I don't think `//` will work

Implementation detail:
https://docs.python.org/3/library/configparser.html#customizing-parser-behaviour
> comment_prefixes, default value: ('#', ';')

https://github.com/platformio/platformio-core/blob/3ef96cb215024709f463d5b1b5574ad66bca3ebf/platformio/project/config.py#L70 for value lists
https://github.com/platformio/platformio-core/blob/3ef96cb215024709f463d5b1b5574ad66bca3ebf/platformio/project/config.py#L94 for other stuff

File: code/espurna/config/custom.h.example (hunk 1 above)
Comment (davebuk) [reply code/espurna/config/custom.h.example]: Yes, your version makes more sense. I was trying to say the the name in brackets `( MY_SONOFF_BUILD01)` has to be same as the `src_build_flags` name in platformio.override.ini

File: code/platformio_override.ini.example (hunk 3 above)
Comment (davebuk) [reply code/platformio_override.ini.example]: I have got my comment prefixes muddled between the `.h` and `.ini` files. :-)

File: code/platformio_override.ini.example (hunk 2 above)
Comment (davebuk) [reply code/platformio_override.ini.example]: Again yes, that makes more sense.

**Review by davebuk — COMMENTED** [review]
Two files updated.

**Pull request merged**